File: ts/worker.ts
import { Resonate, type Context } from "@resonatehq/sdk";
import type { Yieldable } from "@resonatehq/sdk/dist/src/types";

const resonate = Resonate.remote({group: "foo_nodes", url:"http://localhost:8001"})

function* foo(ctx: Context, identifier: string): Generator<Yieldable, string, any> {
  return (yield* ctx.run(bar, identifier));
}

function bar(_: Context, identifier: string): string {
  return `id: ${identifier}`;
}
resonate.register( foo)

console.log("worker running...")

File: py/worker.py
from collections.abc import Generator
from typing import Any
from resonate import Resonate, Context, Yieldable
from threading import Event
# Connect to remote Resonate worker group
resonate = Resonate.remote(group="foo_nodes", host="http://localhost")

def foo(ctx: Context, identifier: str) -> Generator[Yieldable, Any, str]:
    result = yield ctx.run(bar, identifier)
    return result

# Simple helper function
def bar(ctx: Context, identifier: str) -> str:
    return f"id: {identifier}"

# Register the generator function
resonate.register(foo)

resonate.start()
print("worker running...")


try:
    Event().wait()
except KeyboardInterrupt:
    ...
